src/Lecture2.hs

```diff
@@ -97,7 +117,8 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+dropSpaces :: String -> String
+dropSpaces = takeWhile (not . isSpace) . dropWhile isSpace
```

Comment:
Perfect 🏆

Reply:
Thanks !

Reply:
@chshersh One doubt I have is, how does this efficiently handle infinite trailing spaces? I noticed that there is a testcase for that. Without looking at all the infinite trailing spaces, how can this work? Is it some kind of fusion or something like that?

Reply:
There's no special magic 🙂 
`takeWhile` will just stop on the first space character and won't traverse the list any further. It works due to laziness.

Reply:
Got it, thanks!